### PR TITLE
fix(tianmu): fix Load DPN when the table is empty (#964) 

### LINKED ISSUE #964: bug: Query ERROR 1105 (HY000): assert failed on i < m_idx.size() at rc_attr.h:342
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```shell
mysql> select t1.id,t1.name,t2.id,t2.name from t1 left join t2 on t1.name=t2.name where t1.name in ('abc');
ERROR 1105 (HY000): assert failed on i < m_idx.size() at rc_attr.h:342, msg: [bad dpn index 0/0]
```

### Expected behavior

```shell
mysql> select t1.id,t1.name,t2.id,t2.name from t1 left join t2 on t1.name=t2.name where t1.name in ('abc');
+------+------+------+------+
| id   | name | id   | name |
+------+------+------+------+
|    1 | abc  | NULL | NULL |
+------+------+------+------+
1 row in set (0.00 sec)
```

### How To Reproduce

```sql
CREATE TABLE `t1` (
  `id` int(11) DEFAULT NULL,
  `name` varchar(20) DEFAULT NULL,
  KEY `idx_n` (`name`)
) DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
CREATE TABLE `t2` (
  `id` int(11) DEFAULT NULL,
  `name` varchar(20) DEFAULT NULL,
  KEY `idx_n` (`name`)
)DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

insert into t1 values(1,'abc');
insert into t1 values(2,'def');

select t1.id,t1.name,t2.id,t2.name from t1 left join t2 on t1.name=t2.name where t1.name in ('abc');
```

### Environment

```shell
root@localhost:/# /stonedb57/install/bin/mysqld --version
/stonedb57/install/bin/mysqld  Ver 5.7.36-StoneDB for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: d0c2e01b6
        Last commit time: Date:   Wed Nov 2 19:58:00 2022 +0800
        Build time: Date: Sun 06 Nov 2022 08:50:06 AM UTC
root@ub01:~# cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by adofsauron**
## Summary about this PR

Issue Number: close #964 

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Review by RingsC — APPROVED** [review]
LGTM

File: storage/tianmu/core/rc_attr.cpp [hunk 1]
```diff
@@ -1,1429 +1,1436 @@
...
-    if (returnCode == common::ErrorCode::FAILED) {
-      TIANMU_LOG(LogCtl_Level::DEBUG, "Delete: %" PRId64 " for primary key", currentValue);
-      throw common::Exception("Delete: " + std::to_string(currentValue) + " for primary key");
-    }
-  }
-}
-
-}  // namespace core
-}  // namespace Tianmu
+/* Copyright (c) 2022 StoneAtom, Inc. All rights reserved.
```
Comment (lujiashun) [inline storage/tianmu/core/rc_attr.cpp]: File format changes betweent DOS and UNIX formart. Can not distingish what is the differ. Pls just commit the changed lines. You can use unix2dos or dos2unix to keep the file format consistency.

Comment (adofsauron) [reply storage/tianmu/core/rc_attr.cpp]: in the future this would be take of

**Review by lujiashun — APPROVED** [review]
LGTM

**Pull request merged**